### ISSUE TITLE
Compiling Apollo Causes Segfault

### ISSUE THREAD

**Issue by GoogleCodeExporter** [cwalls251/iphone-dev#2]
```
Compiling it causes issues.

Apollo .1 svn Revision: 102


ApolloIMApp.m: In function '-[ApolloIMApp
applicationIsReadyToSuspend]':ApolloIMApp.m:107: internal compiler error:
Segmentation faultPlease submit a full bug report,with preprocessed source
if appropriate.See <URL:http://developer.apple.com/bugreporter> for
instructions.make: *** [ApolloIMApp.o] Error 1
```

Original issue reported on code.google.com by `mikelike...@gmail.com` on 11 Sep 2007 at 7:28

**Comment by GoogleCodeExporter** [comment]
Original comment by `nightwat...@gmail.com` on 11 Sep 2007 at 8:32
- Added labels: **Priority-High**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
Original comment by `nightwat...@gmail.com` on 11 Sep 2007 at 9:51
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
Original comment by `nightwat...@gmail.com` on 11 Sep 2007 at 10:11
- Changed state: **Forwarded**

**Comment by GoogleCodeExporter** [comment]
```
Running delta against this to see what's wrong.
```

Original comment by `nightwat...@gmail.com` on 11 Sep 2007 at 10:12
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
The following reproduces the bug:

-----

#import <Foundation/Foundation.h>

@interface FooApp : NSObject

@end

@implementation FooApp

- (void)foo
{
    NSLog(@"foo");
    NSLog(@"bar"); 
}

@end

-----

Forwarded. In the meantime, if LLVM-GCC 4.2 looks to be this unstable from the 
outset, we're screwed. We 
need to downgrade to LLVM-GCC 4.0 for now and go with that for the time being.


```

Original comment by `nightwat...@gmail.com` on 12 Sep 2007 at 1:11

**Comment by GoogleCodeExporter** [comment]
```
You can produce the bug without even calling NSLog.

just NSString *foo = @"Foo"; etc will do it.
```

Original comment by `mikelike...@gmail.com` on 12 Sep 2007 at 1:13

**Comment by GoogleCodeExporter** [comment]
```
Should be fixed, in that we dropped back to LLVM-GCC 4.0 until 4.2 is more 
stable.
```

Original comment by `nightwat...@gmail.com` on 12 Sep 2007 at 4:54
- Changed state: **Fixed**
